### PR TITLE
Add functions to test images

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: c2cwsgiutils/acceptance/image.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+def check_image_file(
+    result_folder, image_filename_to_check, image_filename, level=0.99, generate_image: bool = False
+):
+    result = skimage.io.imread(image_filename_to_check)
+    assert result is not None, "Wrong image: " + image_filename_to_check
+    check_image(result_folder, result, image_filename, level, generate_image)
+
+
+def check_image(result_folder, image_to_check, image_filename, level=0.99, generate_image: bool = False):
```
Comment (gberaudo) [inline c2cwsgiutils/acceptance/image.py:28]: It looks strange to use __file__ to resolve where the expected image is located. I think the caller should  know and provide full path.

**Review by gberaudo — APPROVED** [review]

File: tests/test_acceptance_image.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+def test_good():
+    image = os.path.join(os.path.dirname(__file__), "test.expected.png")
+    check_image_file("/results", image, os.path.join(os.path.dirname(__file__), "test.expected.png"))
+
+
+def test_wrong():
+    image = os.path.join(os.path.dirname(__file__), os.path.join(os.path.dirname(__file__), "test.wrong.png"))
+    with pytest.raises(AssertionError):
+        check_image_file("/results", image, "test")
+        assert False, "Expected assertion error"
```
Comment (arnaud-morvan) [inline tests/test_acceptance_image.py:17]: Not sure you need this last line.

File: c2cwsgiutils/acceptance/image.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    result_folder: str,
+    image_filename_to_check: str,
+    expected_filename: str,
+    level: float = 0.99,
+    generate_image: bool = False,
+) -> None:
+    """
+    Test that the `<image_filename_to_check>` corresponds to the image `tests/<image_filename>.expected.png`.
+
+    If they don't corresponds the images `<result_folder>`/<image_filename>.result.png and
```
Comment (arnaud-morvan) [inline c2cwsgiutils/acceptance/image.py:26]:
```suggestion
    If the images don't match,  `<result_folder>`/<image_filename>.result.png and
```

File: c2cwsgiutils/acceptance/image.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    `<result_folder>`/<image_filename>.diff.png are created with the corresponding content.
+
+    Where is the name if the `expected_filename`.
+    """
+    assert image_to_check is not None, "Image required"
+    image_filename = os.path.splitext(os.path.basename())[0]
+    if image_filename.endswith(".expected"):
+        image_filename = os.path.splitext(image_filename)[0]
+    result_name = os.path.join(result_folder, f"{image_filename}.result.png")
```
Comment (arnaud-morvan) [inline c2cwsgiutils/acceptance/image.py:55]: image_filename is a basename
but
result_name is a path
and
expected_filename is also a path.

Desserve some consistency on variable names.

File: c2cwsgiutils/acceptance/image.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        skimage.io.imsave(result_name, image_to_check)
+        skimage.io.imsave(expected_filename, image_to_check)
+        assert False, "Expected image not found: " + expected_filename
+    expected = skimage.io.imread(expected_filename)
+    assert expected is not None, "Wrong image: " + expected_filename
+    score, diff = _image_diff(expected, image_to_check)
+    if diff is None:
+        skimage.io.imsave(result_name, image_to_check)
+        assert diff is not None, "No diff generated"
+    if score > level:
```
Comment (arnaud-morvan) [inline c2cwsgiutils/acceptance/image.py:70]: Are you sure we raise an error if score > level ?
level defaults to 0.99

File: c2cwsgiutils/acceptance/image.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+    Where is the name if the `expected_filename`.
+    """
+    assert image_to_check is not None, "Image required"
+    image_filename = os.path.splitext(os.path.basename())[0]
+    if image_filename.endswith(".expected"):
+        image_filename = os.path.splitext(image_filename)[0]
+    result_name = os.path.join(result_folder, f"{image_filename}.result.png")
+    diff_name = os.path.join(result_folder, f"{image_filename}.diff.png")
+    if generate_image:
```
Comment (arnaud-morvan) [inline c2cwsgiutils/acceptance/image.py:57]:
```suggestion
    if overwrite_expected:
```

File: c2cwsgiutils/acceptance/image.py (hunk 3 above)
Comment (arnaud-morvan) [reply c2cwsgiutils/acceptance/image.py:26]: Or maybe:

```suggestion
    If the images differ too much,  `<result_folder>`/<image_filename>.result.png and
```

File: c2cwsgiutils/acceptance/image.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    level: float = 0.99,
+    generate_image: bool = False,
+) -> None:
+    """
+    Test that the `<image_filename_to_check>` corresponds to the image `tests/<image_filename>.expected.png`.
+
+    If they don't corresponds the images `<result_folder>`/<image_filename>.result.png and
+    `<result_folder>`/<image_filename>.diff.png are created with the corresponding content.
+
+    Where is the name if the `expected_filename`.
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply c2cwsgiutils/acceptance/image.py:29]: Oups: => 
Where `<image_filename>` is the name of the `expected_filename`.

File: c2cwsgiutils/acceptance/image.py (hunk 3 above)
Comment (sbrunner) [reply c2cwsgiutils/acceptance/image.py:26]: +1 :-)

File: c2cwsgiutils/acceptance/image.py (hunk 4 above)
Comment (sbrunner) [reply c2cwsgiutils/acceptance/image.py:55]: +1 => 
image_file_basename
result_filename
diff_filename

File: c2cwsgiutils/acceptance/image.py (hunk 5 above)
Comment (sbrunner) [reply c2cwsgiutils/acceptance/image.py]: No, I effectively see that, fixed :-)

File: c2cwsgiutils/acceptance/image.py (hunk 6 above)
Comment (sbrunner) [reply c2cwsgiutils/acceptance/image.py:57]: I prefer generate_expected_image and I deserve a comment

File: tests/test_acceptance_image.py (hunk 2 above)
Comment (sbrunner) [reply tests/test_acceptance_image.py:17]: +1 => remove :-)

File: tests/test_acceptance_image.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+from c2cwsgiutils.acceptance.image import check_image_file
+
+
+def test_good():
+    image = os.path.join(os.path.dirname(__file__), "test.expected.png")
+    check_image_file("/results", image, os.path.join(os.path.dirname(__file__), "test.expected.png"))
+
+
+def test_wrong():
+    image = os.path.join(os.path.dirname(__file__), os.path.join(os.path.dirname(__file__), "test.wrong.png"))
```
Comment (arnaud-morvan) [inline tests/test_acceptance_image.py:14]: @sbrunner 

This line looks wreid:
```
os.path.join(dirname(__file__), os.path.join(dirname(__file__), ...))
```
I don't know how this test can pass.

Comment (sbrunner) [reply tests/test_acceptance_image.py:14]: os.path.join('/tmp/test', '/toto/truc')
=>
'/toto/truc'
Then its working :-)

I will fix that...

**Pull request merged**